src/main/java/org/jenkinsci/plugins/workflow/support/steps/FilePathDynamicContext.java

```diff
@@ -66,6 +68,13 @@
                     e.addSuppressed(tr);
                 }
             }
+            TaskListener listener = context.get(TaskListener.class);
+            if (listener != null) {
+                OfflineCause oc = c.getOfflineCause();
+                if (oc != null) {
+                    listener.getLogger().println(c.getDisplayName() + " was marked offline: " + oc);
```

Comment:
Note that this assumes `OfflineCause.toString` has been meaningfully overridden. That is definitely true of `SimpleOfflineCause`, and seems to be done for other subtypes as well, though really it ought to have been defined as `abstract` to ensure that it is implemented.